src/Microsoft.Framework.FileSystemGlobbing/Internal/MatcherContext.cs

```diff
@@ -53,13 +63,13 @@ private void Match(DirectoryInfoBase directory, string parentRelativePath)
             }
             else
             {
-                foreach (var literal in _declaredLiteralFolderSegments)
+                var candidates = directory.EnumerateFileSystemInfos().Where(info => info is DirectoryInfoBase).Cast<DirectoryInfoBase>();
```

Comment:
The `directory` is of type `DirectoryInfoBase` which doesn't have `EnumerateDirectories` API. (It's not `DirectoryInfo`.